volatility/framework/plugins/windows/dumpfiles.py

```diff
@@ -0,0 +1,245 @@
+            vollog.log(constants.LOGLEVEL_VVV,
+                       "The file object at {0:#x} is not a file on disk".format(file_obj.vol.offset))
+            return
+
+        # Depending on the type of object (DataSection, ImageSection, SharedCacheMap) we may need to
+        # read from the memory layer or the primary layer.
+        memory_layer = self.context.layers["memory_layer"]
+        primary_layer = self.context.layers[self.config["primary"]]
+
+        obj_name = file_obj.file_name_with_device()
+
+        # This stores a list of tuples, describing what to dump and how to dump it.
+        # Ex: (
+        #     memory_object with get_available_pages() API (either CONTROL_AREA or SHARED_CACHE_MAP),
+        #     layer to read from,
+        #     file extension to apply,
+        #     )
+        dump_parameters = []
+
+        # The DataSectionObject and ImageSectionObject caches are handled in basically the same way.
+        # We carve these "pages" from the memory_layer.
+        for member_name, extension in [("DataSectionObject", "dat"), ("ImageSectionObject", "img")]:
+            try:
+                section_obj = getattr(file_obj.SectionObjectPointer, member_name)
+                control_area = section_obj.dereference().cast("_CONTROL_AREA")
+                if control_area.is_valid():
+                    dump_parameters.append((control_area, memory_layer, extension))
+            except exceptions.InvalidAddressException:
+                vollog.log(constants.LOGLEVEL_VVV,
+                           "{0} is unavailable for file {1:#x}".format(member_name, file_obj.vol.offset))
+
+        # The SharedCacheMap is handled differently than the caches above.
+        # We carve these "pages" from the primary_layer.
+        try:
+            scm_pointer = file_obj.SectionObjectPointer.SharedCacheMap
+            shared_cache_map = scm_pointer.dereference().cast("_SHARED_CACHE_MAP")
+            if shared_cache_map.is_valid():
+                dump_parameters.append((shared_cache_map, primary_layer, "vacb"))
+        except exceptions.InvalidAddressException:
+            vollog.log(constants.LOGLEVEL_VVV,
+                       "SharedCacheMap is unavailable for file {0:#x}".format(file_obj.vol.offset))
+
+        for memory_object, layer, extension in dump_parameters:
+            cache_name = EXTENSION_CACHE_MAP[extension]
+            desired_file_name = "file.{0:#x}.{1:#x}.{2}.{3}.{4}".format(file_obj.vol.offset,
+                                                                        memory_object.vol.offset,
+                                                                        cache_name,
+                                                                        ntpath.basename(obj_name),
+                                                                        extension)
+
+            result_text = self.dump_file_producer(file_obj, memory_object, layer, desired_file_name)
```

Comment:
As mentioned, I'd return a `FileInterface` object here, and if you still want specific text (even though it's effectively a boolean flag) then I'd return a tuple here.  (Personally, I'd prefer the boolean based on whether FileInterface is None, but I can live with either).  5:)